info-compiler/src/test/java/com/google/sps/InfoCompilerTest.java

```diff
@@ -0,0 +1,239 @@
+    datastoreHelper = LocalDatastoreHelper.create();
+    datastoreHelper.start();
+    datastore = datastoreHelper.getOptions().getService();
+    infoCompiler = new InfoCompiler(datastore);
+
+    ELECTION_JSON = new JsonObject();
+    ELECTION_JSON.addProperty("kind", "civicinfo#electionsqueryresponse");
+    JsonArray elections = new JsonArray(1);
+    JsonObject election = new JsonObject();
+    election.addProperty("id", "2000");
+    election.addProperty("name", "VIP Test Election");
+    election.addProperty("electionDay", "2013-06-06");
+    elections.add(election);
+    ELECTION_JSON.add("elections", elections);
+
+    SINGLE_CONTEST_JSON = new JsonObject();
+    SINGLE_CONTEST_JSON.addProperty("office", "Governor");
+    JsonArray candidates = new JsonArray(1);
+    JsonObject candidate = new JsonObject();
+    candidate.addProperty("name", "Andrew Cuomo");
+    candidate.addProperty("party", "Democratic");
+    candidates.add(candidate);
+    SINGLE_CONTEST_JSON.add("candidates", candidates);
+  }
+
+  /**
+   * Resets the internal state of the Datastore emulator and then {@code datastore}. Also resets
+   * {@code infoCompiler}. We choose to reset, instead of creating/destroying the Datastore
+   * emulator at each test, because {@code datastoreHelper.stop()} sometimes generates a {@code
+   * java.net.ConnectException}, when making HTTP requests. Hence we try to limit the number of
+   * times {@code datastoreHelper} is created/destroyed.
+   */
+  @Before
+  public void resetDatastore() throws IOException {
+    datastoreHelper.reset();
+    datastore = datastoreHelper.getOptions().getService();
+    infoCompiler = new InfoCompiler(datastore);
+  }
+
+  @Test
+  public void queryCivicInformation_succeedWithMockResponse() throws Exception {
+    // Query the Civic Information API with a mock HTTP client + mock callback function of type
+    // {@code ResponseHandler<String>} that converts any {@code HttpResponse} response to {@code
+    // ELECTION_RESPONSE}, and subsequently convert {@code ELECTION_RESPONSE} to {@code json}.
+    // {@code httpGet} is irrelevant in this test, since {@code execute()} is mocked as above.
+    // Since String {@code ELECTION_RESPONSE} and JsonObject {@code ELECTION_JSON} match in
+    // content, {@code json} should be exactly the same as {@code ELECTION_JSON}. Here, we don't
+    // repeatedly test the same thing with {@code SINGLE_CONTEST_JSON}.
+    CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+    HttpGet httpGet = new HttpGet(ELECTION_QUERY_URL);
+    ArgumentCaptor<ResponseHandler<String>> argumentCaptor =
+        ArgumentCaptor.forClass(ResponseHandler.class);
+    when(httpClient.execute(anyObject(), argumentCaptor.capture())).thenReturn(ELECTION_RESPONSE);
+    JsonObject json =
+        infoCompiler.requestHttpAndBuildJsonResponseFromCivicInformation(httpClient, httpGet);
+    assertThat(ELECTION_JSON).isEqualTo(json);
```

Comment:
*Sorry actually for Truth the ordering is opposite (i.e. you want to assert things about `json`, such that it is equal to the expected `ELECTION_JSON`)

```suggestion
    assertThat(json).isEqualTo(ELECTION_JSON);
```